### PR TITLE
Oy2 19467 - Footer component

### PR DESCRIPTION
**Pull request by jdinh8124**
## Purpose
- Create react footer component based on figma designs and USWDS
Resolves:
https://qmacbis.atlassian.net/jira/software/c/projects/OY2/boards/240?modal=detail&selectedIssue=OY2-19467

#### Pull Request Creator Checklist

- [x] It is documented what type of request this is.
- [x] There is a detailed description for the purpose of this PR.
- [x] This PR is in line with the intention of this Project

### CODE REVIEW

File: src/components/Footer/Footer.tsx [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+import { Icon, IconChoice } from "../Icon/Icon";
+type IntrinsicElements = JSX.IntrinsicElements["footer"];
+
+interface Link {
+  href?: string;
+  iconName?: IconChoice;
+  onClick?: () => any;
+  text: string;
+}
+
```
Comment (jdinh8124) [inline src/components/Footer/Footer.tsx:17]: I see JP is having an incoming interface thats similar maybe we can export one of them

File: src/components/Footer/Footer.tsx [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+type IntrinsicElements = JSX.IntrinsicElements["footer"];
+
+interface Link {
+  href?: string;
+  iconName?: IconChoice;
+  onClick?: () => any;
+  text: string;
+  target?: string;
+}
+
```
Comment (jdinh8124) [inline src/components/Footer/Footer.tsx:17]: I see JP is having an incoming interface thats similar maybe we can export one of them

File: src/components/Footer/Footer.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    address: {
+      description: "Address used at the bottom right of footer",
+    },
+    altFooter: {
+      description: "Determines whether to use the alternative footer",
+    },
+    navigationLinks: {
+      description:
+        "Array of navigation elements to render in the alternative footer",
+    },
```
Comment (jphechter) [inline src/components/Footer/Footer.stories.tsx:21]:
```suggestion
    navigationLinks: {
      description:
        "Array of navigation elements to render in the alternative footer",
        control: false,
    },
```
I think this is a good candidate for control to disable in Storybook.

**Review by jphechter — CHANGES_REQUESTED** [review]
1 note about the story implementation, other than that LGTM 👍 

I did run the tests locally just to check coverage. Noticed that the images added in this PR are getting picked up for some reason. I tried updating `jest.config.ts` adding `assets` to both `coveragePathIgnorePatterns` and `testPathIgnorePatterns`. Neither seemed to work.
Screen Shot 2022-08-10 at 2 28 55 PM
 Their coverage is at 100% 🙃 so I don't really see an issue. Just something interesting.

**Review by jphechter — APPROVED** [review]

**Pull request merged**